modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -81,13 +97,70 @@ export default function AuditBarChart({rtl, vocabularies}) {
 		? [BAR_CHART.dotRadiusMin]
 		: [BAR_CHART.legendHeight + BAR_CHART.dotRadiusMin];
 
+	const [checkboxes, setCheckbox] = useState(legendCheckboxes);
+
+	useEffect(() => {
```

Comment:
One thing to consider is that [`useLayoutEffect`](https://reactjs.org/docs/hooks-reference.html#uselayouteffect) *might* be a better choice here. It probably doesn't matter because I don't think this component is likely to update often, but if it did, there is a chance that `useLayoutEffect` might avoid a repaint, because:

> Updates scheduled inside `useLayoutEffect` will be flushed synchronously, before the browser has a chance to paint.

In other words, you update the chart DOM, you inject the `style` element, and *then* the browser repaints (once), instead of updating chart, layout/painting, injecting style element, repainting. But like I said, unless the component is rendering in a loop, you are unlikely to notice a difference. Just wanted to bring it up though as food for thought.

Reply:
I'm controlling the show/hide bars with the checkbox in the legend with a state in the component but probably in the future, we want that when the user filters the list, affects also in the chart and for that reason maybe `useLayoutEffect` it's a better choice. What do you think?

Reply:
By "frequent updates" I mean something like I said above (eg. "rendering in a loop", something that could happen many times a second, while dragging or scrolling or typing, for instance); something like toggling a checkbox is going to be very infrequent, and limited by the user's ability to click (how many times can you toggle a checkbox in a second? Five?).

I would say if you don't see any visible flicker on render, don't worry about `useLayoutEffect` yet.

Reply:
👍

Reply:
> (how many times can you toggle a checkbox in a second? Five?)

I dare you... I double dare you...

[My Score: 12 CPS](https://clickspeedtest.com/clicks-per-second.html)

![just-sayin](https://user-images.githubusercontent.com/905006/85711309-7089c880-b6e7-11ea-8bd3-aab7cc6fb923.gif)